cluster/common.sh

```diff
@@ -353,3 +353,40 @@ function find-release-version() {
     exit 1
   fi
 }
+
+function stage-images() {
+  find-release-version
+  find-release-tars
+
+  KUBE_IMAGE_TAG="$(echo """${KUBE_GIT_VERSION}""" | sed 's/+/-/g')"
+
+  local docker_wrapped_binaries=(
+    "kube-apiserver"
+    "kube-controller-manager"
+    "kube-scheduler"
+    "kube-proxy"
+  )
+
+  local docker_cmd=("docker")
+
+  if [[ "${KUBE_DOCKER_REGISTRY}" == "gcr.io/"* ]]; then
+    local docker_push_cmd=("gcloud" "docker")
+  fi
+
+  local temp_dir="$(mktemp -d -t 'kube-server-XXXX')"
+
+  tar xzfv "${SERVER_BINARY_TAR}" -C "${temp_dir}" &> /dev/null
+
+  for binary in "${docker_wrapped_binaries[@]}"; do
+    local docker_tag="$(cat ${temp_dir}/kubernetes/server/bin/${binary}.docker_tag)"
+    (
+      "${docker_cmd[@]}" load -i "${temp_dir}/kubernetes/server/bin/${binary}.tar"
+      "${docker_cmd[@]}" tag -f "gcr.io/google_containers/${binary}:${docker_tag}" "${KUBE_DOCKER_REGISTRY}/${binary}:${KUBE_IMAGE_TAG}"
+      "${docker_push_cmd[@]}" push "${KUBE_DOCKER_REGISTRY}/${binary}:${KUBE_IMAGE_TAG}"
+    ) &> /dev/null &
+  done
+
+  wait
```

Comment:
I think you should probably use the [wait-for-jobs](https://github.com/kubernetes/kubernetes/blob/794f162ef0b0f5e9e9555f923405f3120df10f01/hack/lib/util.sh#L178) pattern here, and you may want to dump the output into a file in `${temp_dir}` rather than `/dev/null` in case there is an error.